transcend/data_source_sombra.go

```diff
@@ -0,0 +1,73 @@
+package transcend
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	graphql "github.com/hasura/go-graphql-client"
+	"github.com/transcend-io/terraform-provider-transcend/transcend/types"
+)
+
+func dataSourceSombra() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceSombraRead,
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"url": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The url to lookup a sombra instance by",
+			},
+		},
+	}
+}
+
+func dataSourceSombraRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*Client)
+
+	var diags diag.Diagnostics
+
+	var query struct {
+		Sombras []types.Sombra `graphql:"sombras(filterBy: { urls: [$url] })"`
+	}
+
+	vars := map[string]interface{}{
+		"url": graphql.String(d.Get("url").(string)),
```

Comment:
I'm wondering if a more specific type than `interface{}` can be used here. `graphql.String(d.Get("url").(string))` would make me think it returns a `string`. Is that true?

Reply:
Types are the thing that confuse me the most about go...but I saw this pattern in the other data_source so I copied pasta...

Reply:
I notice you had to explicitly use pointers in this. I'm sorry